File: tests/test_localization_switching.py
import pytest
from ui import locales


def test_localization_switching_branding():
    """Verify that switching locales updates strings and maintains ZenAI branding."""
    supported_codes = ["en", "es", "fr", "ro", "hu", "he"]

    for code in supported_codes:
        # Switch locale
        loc = locales.set_locale(code)

        # Verify language code matches
        assert locales.get_locale_code() == code

        # Verify branding consistency across all languages
        # These should NEVER be "ZenAI" or "ZenAI" (in user-facing content)
        assert "ZenAI" in loc.APP_NAME
        assert "ZenAI" in loc.APP_TITLE

        # Check welcome message key exists (we renamed it to WELCOME_ZENAI)
        assert hasattr(loc, "WELCOME_ZENAI")
        assert not hasattr(loc, "WELCOME_ZENA")

        # Check chat placeholder key exists
        assert hasattr(loc, "CHAT_PLACEHOLDER_ZENAI")
        assert not hasattr(loc, "CHAT_PLACEHOLDER_ZENA")


def test_invalid_locale_raises_error():
    """Verify that an unsupported locale code raises ValueError."""
    with pytest.raises(ValueError):
        locales.set_locale("zz_ZZ")


def test_shorthand_access():
    """Verify that L() shorthand works correctly after a switch."""
    locales.set_locale("hu")
    assert locales.L().LANGUAGE_CODE == "hu"

    locales.set_locale("en")
    assert locales.L().LANGUAGE_CODE == "en"


def test_menu_strings_not_empty():
    """Verification that critical UI strings are translated (not empty)."""
    loc = locales.set_locale("ro")
    assert loc.NAV_MODEL_MANAGER
    assert loc.BTN_SEND
    assert loc.CHAT_PLACEHOLDER

File: rust_output/src/test_localization_switching.rs
use anyhow::{Result, Context};

/// Verify that switching locales updates strings and maintains ZenAI branding.
pub fn test_localization_switching_branding() -> () {
    // Verify that switching locales updates strings and maintains ZenAI branding.
    let mut supported_codes = vec!["en".to_string(), "es".to_string(), "fr".to_string(), "ro".to_string(), "hu".to_string(), "he".to_string()];
    for code in supported_codes.iter() {
        let mut loc = locales.set_locale(code);
        assert!(locales.get_locale_code() == code);
        assert!(loc.APP_NAME.contains(&"ZenAI".to_string()));
        assert!(loc.APP_TITLE.contains(&"ZenAI".to_string()));
        assert!(/* hasattr(loc, "WELCOME_ZENAI".to_string()) */ true);
        assert!(!/* hasattr(loc, "WELCOME_ZENA".to_string()) */ true);
        assert!(/* hasattr(loc, "CHAT_PLACEHOLDER_ZENAI".to_string()) */ true);
        assert!(!/* hasattr(loc, "CHAT_PLACEHOLDER_ZENA".to_string()) */ true);
    }
}

/// Verify that an unsupported locale code raises ValueError.
pub fn test_invalid_locale_raises_error() -> () {
    // Verify that an unsupported locale code raises ValueError.
    let _ctx = pytest.raises(ValueError);
    {
        locales.set_locale("zz_ZZ".to_string());
    }
}

/// Verify that L() shorthand works correctly after a switch.
pub fn test_shorthand_access() -> () {
    // Verify that L() shorthand works correctly after a switch.
    locales.set_locale("hu".to_string());
    assert!(locales.L().LANGUAGE_CODE == "hu".to_string());
    locales.set_locale("en".to_string());
    assert!(locales.L().LANGUAGE_CODE == "en".to_string());
}

/// Verification that critical UI strings are translated (not empty).
pub fn test_menu_strings_not_empty() -> () {
    // Verification that critical UI strings are translated (not empty).
    let mut loc = locales.set_locale("ro".to_string());
    assert!(loc.NAV_MODEL_MANAGER);
    assert!(loc.BTN_SEND);
    assert!(loc.CHAT_PLACEHOLDER);
}
